### PR TITLE
Add OsinServerConfig

### PR DESCRIPTION
**Pull request by enj**
This change adds the CLI configuration for use by the standalone OAuth server.

Signed-off-by: Monis Khan <mkhan@redhat.com>

@openshift/sig-auth 

/assign @derekwaynecarr

### CODE REVIEW

File: osin/v1/types.go [hunk 1]
```diff
@@ -7,17 +7,56 @@ import (
...
+	// loginURL is the URL required to login to the cluster:
+	// oc login --server=<loginURL>
+	// masterPublicURL is the issuer URL
+	// it is accessible from inside (service network) and outside (ingress) of the cluster
+	// masterURL is the loopback variation of the token_endpoint URL with no path component
+	// it is only accessible from inside (service network) of the cluster
+	// masterCA is used to perform TLS verification for connections made to masterURL
+	// For further details, see the IETF Draft:
+	// https://tools.ietf.org/html/draft-ietf-oauth-discovery-04#section-2
+	LoginURL string `json:"loginURL"`
```
Comment (ericavonb) [inline osin/v1/types.go]: Should we be keep consistent with the oauth spec and call this like `AuthorizationEndpoint`?

File: osin/v1/types.go [hunk 2]
```diff
@@ -7,17 +7,56 @@ import (
...
+	// loginURL, along with masterCA, masterURL and masterPublicURL have distinct
+	// meanings depending on how the OAuth server is run.  The two states are:
+	// 1. embedded in the kube api server (all 3.x releases)
+	// 2. as a standalone external process (all 4.x releases)
+	// in the embedded configuration, loginURL is equivalent to masterPublicURL
+	// and the other fields have functionality that matches their docs.
+	// in the standalone configuration, the fields are used as:
+	// loginURL is the URL required to login to the cluster:
+	// oc login --server=<loginURL>
+	// masterPublicURL is the issuer URL
```
[the remark this answers is not in the document]
Comment (enj) [reply osin/v1/types.go]: That is not what this is.  `issuer` is a specific field in the oauth metadata.

File: osin/v1/types.go [hunk 3]
```diff
@@ -7,17 +7,56 @@ import (
...
 	// masterURL is used for making server-to-server calls to exchange authorization codes for access tokens
+	// This field is deprecated and will be removed in a future release.
+	// See loginURL for details.
+	// Deprecated
 	MasterURL string `json:"masterURL"`
 
 	// masterPublicURL is used for building valid client redirect URLs for internal and external access
+	// This field is deprecated and will be removed in a future release.
+	// See loginURL for details.
+	// Deprecated
```
[the remark this answers is not in the document]
Comment (enj) [reply osin/v1/types.go]: I tried to cram it all in the `loginURL` doc.  I do not expect this to ever be clear until we can remove the old path.

File: osin/v1/types.go (hunk 1 above)
Comment (enj) [reply osin/v1/types.go]: This is not the `AuthorizationEndpoint` - see the `oc` command on how it is used.

**Pull request merged**